catalog/core/catalog-core-impl/pubsub/src/main/java/ddf/catalog/pubsub/PublishedEventHandler.java

```diff
@@ -47,89 +54,110 @@ public PublishedEventHandler(Predicate finalPredicate, Subscription subscription
         this.subscription = subscription;
         this.preDelivery = preDelivery;
         this.catalog = catalog;
+        //this will queue tasks to execute 1 task at a time submitting a task is a non blocking call.
+        //the thread is cached but if it is not used for more than 60 seconds it will be destroyed.
```

Comment:
Couldn't we just have the `EventProcessorImpl` class be the one that creates the thread pool and submits an instance of this class to it? That would allow us to create the thread pool once in blueprint and inject it in the `EventProcessorImpl` class. It also looks odd to me to have this class creates the thread pool. Feels to me like the threading decision should be left to the class using it. Thoughts?